### PR TITLE
[TXP-48] Test Pilot Web Experiments

### LINKED ISSUE #2546: TXP-48: Test Pilot Web Experiments
@johngruen 
In order to accommodate web experiments on Test Pilot, I created the proposal and took some future possibilities into consideration as well to ensure the flexibility, such as mobile experiments. However, for the current implementation, please feel free to implement only what we need for now. Besides, for web experiments, perhaps we can display how many users visit the web experiment instead of removing the user counts section. What do you think? 

web experiments

Please find assets here.
web experiment: https://github.com/mozilla/testpilot-assets/blob/master/Test_Pilot_Website/00_Assets/webexperiment.svg
add-on: http://design.firefox.com/icons/icons/addons/extensions.svg
mobile: http://design.firefox.com/icons/icons/new/device-mobile-16.svg

### PR DESCRIPTION
**Pull request by lmorchard**
- New YAML properties:

  - platforms: ["addon", "web", "mobile"]

  - web_url: https://example.com

- New ExperimentPlatforms stateless component

- Add platform indicator to experiment cards and detail page

- Use "Go to" link instead of "Enable" button for web platform experiments

- Updates to docs & tests

Fixes #2546

### CODE REVIEW

File: frontend/src/styles/modules/_experiment-details.scss [hunk 1]
```diff
@@ -435,6 +435,42 @@
...
+    background-image: url('/static/images/experiment-type-addon.svg');
+  }
+
+  .platform-icon-mobile {
+    background-image: url('/static/images/experiment-type-mobile.svg');
+  }
+}
+
+.experiment-information .experiment-platform {
+  margin: ($grid-unit * .5) 0 (($grid-unit * -.5) + 8px);
```
Comment (lmorchard) [inline frontend/src/styles/modules/_experiment-details.scss]: I think my sass-fu has failed me, here

**Comment by lmorchard** [comment]
As a trial, I rebased my Storybook deployment script into this branch.

[Here's a link to a view where all three platforms are turned on in the "Knobs" tab at the bottom](http://storybook.testpilot.dev.mozaws.net/d420f64bc77c6d358fc106c48e04250df26ef8a0/?knob-Has%20Addon=false&knob-Enabled=false&knob-Experiment=%7B%22title%22%3A%22Sample%20experiment%22%2C%22description%22%3A%22This%20is%20an%20example%20experiment%22%2C%22subtitle%22%3A%22%22%2C%22platforms%22%3A%5B%22web%22%2C%22addon%22%2C%22mobile%22%5D%2C%22slug%22%3A%22snooze-tabs%22%2C%22survey_url%22%3A%22https%3A%2F%2Fexample.com%22%2C%22installation_count%22%3A%2286753%22%2C%22created%22%3A%222010-06-21T12%3A12%3A12Z%22%2C%22modified%22%3A%222010-06-21T12%3A12%3A12Z%22%7D&selectedKind=ExperimentRowCard&selectedStory=base%20state%20with%20knobs&full=0&down=1&left=1&panelRight=0&downPanel=storybooks%2Fstorybook-addon-knobs)

**Comment by lmorchard** [comment]
As a further trial, I rebased another tweak to the Storybook deployment to get this link working:

http://storybook.testpilot.dev.mozaws.net/pull/2554/

File: frontend/src/styles/modules/_experiment-details.scss (hunk 1 above)
Comment (johngruen) [reply frontend/src/styles/modules/_experiment-details.scss]: err i read that as `10px 0 -2px` right?

Comment (lmorchard) [reply frontend/src/styles/modules/_experiment-details.scss]: Yeah... that sounds right. Not sure what I was smoking here. I think I was trying to make use of the variables but lost my mind at some point

**Comment by lmorchard** [comment]
Rebased, all todos are done.

**Comment by johngruen** [comment]
@lmorchard i'd like to review and merge this today if possible, but I am not able to access the storybooks you link to above.

File: frontend/src/images/experiment-type-addon.svg [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path d="M14.53,8c-1,0-1,1-1.75,1A.77.77,0,0,1,12,8.25V5a1,1,0,0,0-1-1H7.78A.77.77,0,0,1,7,3.25C7,2.5,8,2.47,8,1.5S7.13,0,6,0,4,.64,4,1.5s1,1,1,1.75A.77.77,0,0,1,4.28,4H1A1,1,0,0,0,0,5V7.25A.77.77,0,0,0,.78,8c.75,0,.78-1,1.75-1S4,7.9,4,9s-.64,2-1.5,2-1-1-1.75-1a.77.77,0,0,0-.75.75V15a1,1,0,0,0,1,1H4.28A.77.77,0,0,0,5,15.25c0-.75-1-.78-1-1.75S4.92,12,6,12s2,.64,2,1.5-1,1-1,1.75a.77.77,0,0,0,.75.75H11a1,1,0,0,0,1-1V11.75a.77.77,0,0,1,.75-.75c.75,0,.78,1,1.75,1s1.5-.9,1.5-2S15.39,8,14.53,8Z" fill="context-fill"/></svg>
```
Comment (johngruen) [inline frontend/src/images/experiment-type-addon.svg]: Should not have fill=`context-fill`, don't set a fill here, or set it to #000

File: frontend/src/images/experiment-type-mobile.svg [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path fill="context-fill" d="M12 0H4a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h8a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2zM9 15H7v-1h2zm3-2.5a.5.5 0 0 1-.5.5h-7a.5.5 0 0 1-.5-.5v-10a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 .5.5z"/></svg>
```
Comment (johngruen) [inline frontend/src/images/experiment-type-mobile.svg]: ditto, re context fill

File: frontend/src/styles/modules/_experiment-details.scss [hunk 4]
```diff
@@ -435,6 +435,42 @@
...
+  font-weight: normal;
+
+  .platform-icon {
+    background-repeat: no-repeat;
+    background-size: 16px;
+    height: 16px;
+    margin: 0;
+    opacity: .5;
+    width: 20px;
+  }
```
Comment (johngruen) [inline frontend/src/styles/modules/_experiment-details.scss]:
```
.experiment-platform {
  @include flex-container(row, flex-start, center);
  color: $transparent-black-5;
  font-size: 1.2 * $font-unit;
  font-weight: normal;

  .platform-icon {
    background-position: left center;
    background-repeat: no-repeat;
    background-size: 16px;
    height: 16px;
    margin-right: 6px;
    margin-top: -4px;
    opacity: .5;
    width: 16px;
  }
```

slight refactor: ^

File: locales/en-US/app.ftl [hunk 5]
```diff
@@ -150,6 +150,14 @@ userCountContainerAlt = Just launched!
 highlightPrivacy = Your privacy
 experimentGradReportPendingTitle = This experiment has ended
 experimentGradReportPendingCopy = We are working on a full report. Check back soon for the details.
+experimentPlatformWebAddonMobile = Available on web / add-on / mobile
+experimentPlatformWebAddon = Available on web / add-on
+experimentPlatformWebMobile = Available on web / mobile
+experimentPlatformAddonMobile = Available on add-on / mobile
+experimentPlatformWeb = Available on web
+experimentPlatformAddon = Available on add-on
+experimentPlatformMobile = Available on mobile
```
Comment (johngruen) [inline locales/en-US/app.ftl]: We should be rewritten as follows:
```
experimentPlatformWebAddonMobile = Firefox / web / mobile experiment
experimentPlatformWebAddon = Firefox / web experiment
experimentPlatformWebMobile = web / mobile experiment
experimentPlatformAddonMobile = Firefox / mobile experiment
experimentPlatformWeb = web experiment
experimentPlatformAddon = Firefox experiment
experimentPlatformMobile = Available on mobile
```

**Review by johngruen — COMMENTED** [review]
@lmorchard looks good! reviewed for strings, svg compat and minor style tweak

File: frontend/src/images/experiment-type-addon.svg (hunk 2 above)
Comment (lmorchard) [reply frontend/src/images/experiment-type-addon.svg]: I just copied these SVG images over from #2546 - I don't know SVG well enough to know what you're asking for here as a change. Are there different images I should use?

Comment (lmorchard) [reply frontend/src/images/experiment-type-addon.svg]: Oh, you mean literally just change from `fill="context-fill"` to `fill="#000"`?

**Pull request merged**